### PR TITLE
Added Active Navbar

### LINKED ISSUE #26: Active nav item styling
The active nav item should be styled differently. Haskell's conditional (`if`) can be used to do this statically without any dynamism.

### PR DESCRIPTION
**Pull request by CaptainLazarus**
resolves #26

### CODE REVIEW

File: src/FPIndia/View.hs [hunk 1]
```diff
@@ -53,12 +53,15 @@ renderBody rp model r = do
         renderMarkdown model "resources.md"
     renderFooter
 
-renderNavbar :: Prism' FilePath Route -> H.Html
-renderNavbar rp =
+renderNavbar :: Prism' FilePath Route -> HtmlRoute -> H.Html
+renderNavbar rp r =
   H.nav ! A.class_ "w-full h-1/4 text-xl font-bold flex space-x-4  mb-4" $ do
-    forM_ universe $ \r -> renderURL (H.toHtml $ routeTitle r) (routeHref rp r)
+    forM_ universe $ \x -> if x == r then renderURL (H.toHtml $ routeTitle x) (routeHref rp x) True else renderURL (H.toHtml $ routeTitle x) (routeHref rp x) False
```
Comment (srid) [inline src/FPIndia/View.hs:59]: `x` is too generic. I'd keep it as `r` and rename the function's argument as `currentRoute`.

File: src/FPIndia/View.hs [hunk 2]
```diff
@@ -53,12 +53,15 @@ renderBody rp model r = do
...
   where
-    renderURL menuItem path = H.a ! A.href path ! A.class_ "p-2 border-b-2 transition color hover:border-stone-900 duration-300" $ menuItem
+    renderURL menuItem path flag =
```
Comment (srid) [inline src/FPIndia/View.hs:61]: rename `flag` -> `active` (or `isActive`).

File: src/FPIndia/View.hs [hunk 3]
```diff
@@ -53,12 +53,15 @@ renderBody rp model r = do
...
+      if flag
+        then H.a ! A.href path ! A.class_ "p-2 border-b-2 border-stone-900" $ menuItem
+        else H.a ! A.href path ! A.class_ "p-2 border-b-2 transition color hover:border-stone-900 duration-300" $ menuItem
```
Comment (srid) [inline src/FPIndia/View.hs:64]: The conditionally is for the class string only, not the entire Html widget. So you want to move the `if` as close to inside as possible. Maybe use `let` block to create a `linkClass` text variable and assign that conditionally.

File: src/FPIndia/View.hs (hunk 1 above)
Comment (srid) [reply src/FPIndia/View.hs]: Or you can rename the nav route argument as `navRoute`.

File: src/FPIndia/View.hs [hunk 4]
```diff
@@ -53,12 +53,21 @@ renderBody rp model r = do
...
+renderNavbar rp currentRoute =
   H.nav ! A.class_ "w-full h-1/4 text-xl font-bold flex space-x-4  mb-4" $ do
-    forM_ universe $ \r -> renderURL (H.toHtml $ routeTitle r) (routeHref rp r)
+    forM_ universe $ \r ->
+      if r == currentRoute
+        then renderURL (H.toHtml $ routeTitle r) (routeHref rp r) True
+        else renderURL (H.toHtml $ routeTitle r) (routeHref rp r) False
   where
-    renderURL menuItem path = H.a ! A.href path ! A.class_ "p-2 border-b-2 transition color hover:border-stone-900 duration-300" $ menuItem
+    renderURL menuItem path isActive =
```
Comment (srid) [inline src/FPIndia/View.hs:64]: This function seems funky to me. 

If you were to add a function comment (not that we should) how would you explain "renderURL"? 

Can that all be made explicit in the function name and type?

Comment (CaptainLazarus) [reply src/FPIndia/View.hs:64]: could rename to 

renderMenuItem title route isActive

File: src/FPIndia/View.hs [hunk 5]
```diff
@@ -53,12 +53,21 @@ renderBody rp model r = do
...
-renderNavbar :: Prism' FilePath Route -> H.Html
-renderNavbar rp =
+renderNavbar :: Prism' FilePath Route -> HtmlRoute -> H.Html
+renderNavbar rp currentRoute =
   H.nav ! A.class_ "w-full h-1/4 text-xl font-bold flex space-x-4  mb-4" $ do
-    forM_ universe $ \r -> renderURL (H.toHtml $ routeTitle r) (routeHref rp r)
+    forM_ universe $ \r ->
+      if r == currentRoute
+        then renderURL (H.toHtml $ routeTitle r) (routeHref rp r) True
+        else renderURL (H.toHtml $ routeTitle r) (routeHref rp r) False
```
Comment (srid) [inline src/FPIndia/View.hs:62]:
```haskell
renderURL (H.toHtml $ routeTitle r) (routeHref rp r) $ r == currentRoute
```

We don't even need the `H.toHtml ` there if you do it in the function itself.

File: src/FPIndia/View.hs [hunk 6]
```diff
@@ -53,12 +53,21 @@ renderBody rp model r = do
...
   where
-    renderURL menuItem path = H.a ! A.href path ! A.class_ "p-2 border-b-2 transition color hover:border-stone-900 duration-300" $ menuItem
+    renderURL menuItem path isActive =
+      let y =
```
Comment (srid) [inline src/FPIndia/View.hs:65]: `y` is too generic a name to use for class names.

File: src/FPIndia/View.hs (hunk 5 above)
Comment (CaptainLazarus) [reply src/FPIndia/View.hs:62]: can actually remove routeHref too I believe. Will change

File: src/FPIndia/View.hs (hunk 6 above)
Comment (CaptainLazarus) [reply src/FPIndia/View.hs:65]: got it.

**Comment by srid** [comment]
I added a new type because a) `NavItem_Active` is more self-documenting than `True` b) separation of concerns in regards to rendering.

**Pull request merged**